matter_server/server/device_controller.py

```diff
@@ -1050,48 +1050,30 @@ async def _call_sdk(self, func: Callable[..., _T], *args: Any, **kwargs: Any) ->
             ),
         )
 
-    async def _check_interview_and_subscription(
-        self, node_id: int, reschedule_interval: int = 30
-    ) -> None:
-        """Handle interview (if needed) and subscription for known node."""
-
+    async def _setup_node(self, node_id: int) -> None:
+        """Handle set-up of subscriptions and interview (if needed) for known/discovered node."""
         if node_id not in self._nodes:
             raise NodeNotExists(f"Node {node_id} does not exist.")
 
         # (re)interview node (only) if needed
-        node_data = self._nodes.get(node_id)
+        node_data = self._nodes[node_id]
         if (
-            node_data is None
+            # re-interview if we dont have any node attributes (empty node)
+            not node_data.attributes
             # re-interview if the schema has changed
-            or node_data.interview_version < SCHEMA_VERSION
+            or node_data.interview_version < DATA_MODEL_SCHEMA_VERSION
+            # re-interview if the last interview was too long ago
+            or (datetime.utcnow() - node_data.last_interview) > timedelta(days=90)
```

Comment:
Any specific idea behind this?

Maybe (another/better) indicator would be software version? :thinking:

Reply:
yeah I dunno, it somehow felt good but at the same time meh... I'll remove it.
software version is tracked elsewhere

Reply:
adjusted